### ISSUE TITLE
[BUG REPORT] `start-at-login` is technically broken, but preventing a worse bug

### ISSUE THREAD

**Issue by jakenvac** [nikitabobko/AeroSpace#479]
The startup at login routine for configuring the launch agent plist & loading/unloading it is technically not working as intended.

### The Bug

Both the commands to [load and unload](https://github.com/nikitabobko/AeroSpace/blob/main/Sources/AppBundle/config/startAtLogin.swift#L27-L31) the plist fail with IO errors as the paths given to them aren't valid - thus the agent is never loaded or unloaded by the app.

`url.absoluteString` keeps the `file://` scheme from the URL instance that is created to locate the plist. launchctl just expects a plain path. To *correct* this behavior `URL.path` should be used instead. However this introduces a messier bug - more on this below.

The reason this behavior still works as well as it does is because the plist is written/deleted when app starts up/reloads config depending on the `start-at-login` value. When you next log in, the plist is automatically loaded by launchd or is missing so cannot be launched.

Running AeroSpaceApp from the terminal shows the following errors:
```sh
# output when the load fails:
Load failed: 5: Input/output error
Try running `launchctl bootstrap` as root for richer errors.

# output when unload fails:
Unload failed: 5: Input/output error
Try running `launchctl bootout` as root for richer errors.
```
You can see slightly more detailed info in the launchd logs:
```sh
# here's one such example for a failed 'load'
2024-08-30 16:58:28.718503 (gui/501 [100018]) <Notice>: Bootstrap by launchctl[21729] for //file:///Users/jake/Library/LaunchAgents/bobko.aerospace.plist failed (2: No such file or directory)
```

### The Messy Bug
When `launchctl load some.plist` is ran and `RunAtLoad` is `true` in the plist, the program is immediately executed - this creates a second instance of AeroSpace and then everything gets wild. Hotkeys are handled twice, windows are moved erratically, and things become generally unpredictable. In my specific scenario, I had things set up in such a way that both instances were fighting over a window and I couldn't focus my terminal to kill the app. Don't ask me how because I'm not sure.

There are many solutions to this problem, such as a seperate startup helper app, creating a lockfile to ensure single instances etc. However I feel this is overkill, at least for a first pass solution.

I think for now, it's safe to remove the calls to launchctl and just rely on the [lack of] presence of the plist and let macos do it's thing on login.

As an aside, truly fixing the launchd integration could bring benefits such as auto restarting the app after an update. #352

Sorry for the verbosity of this bug report. It's been a long day :)

## Checklist

- [X] I've searched for duplicates. My search queries were (please don't forget to search in closed issues too):
  - "log in"
  - "start-at-login"
  - "launchd"
- [ ] I've read the documentation https://nikitabobko.github.io/AeroSpace/guide & https://nikitabobko.github.io/AeroSpace/commands
  - N/A
- [ ] I've searched in the documentation. My search queries were:
  - N/A

```
aerospace CLI client version: 0.14.1-Beta 5861e85b65cb7dc914a0a68027c56e2bc90510fd
AeroSpace.app server version: 0.14.1-Beta 5861e85b65cb7dc914a0a68027c56e2bc90510fd
```
Also happens on main as of 9d8362d04f449c4fdaab0fb2d305e6fe0f4ca48a

**Comment by jakenvac** [comment]
@nikitabobko Are you happy for me to close this now with #480 being merged? Or are you anticipating a more robust solution long term?

**Comment by nikitabobko** [comment]
I prefer to close issues when they are released

**Comment by nikitabobko** [comment]
Released in 0.15.0
